src/dotnet-build/CompileContext.cs

```diff
@@ -275,15 +275,17 @@ private bool InvokeCompileOnDependency(ProjectDescription projectDependency)
 
             args.Add("--framework");
             args.Add($"{projectDependency.Framework}");
+            
+            if (!string.IsNullOrEmpty(_args.RuntimeValue))
```

Comment:
Refactoring idea: Add `ConstructArgumentString` methods to CompilerCommandApp to encapsulate how this string is formed. Then delete the 2 methods from CompileContext